src/Draco.Compiler/Internal/Syntax/Lexer.cs

```diff
@@ -245,6 +246,20 @@ Unit TakeWithText(TokenType tokenType, int length)
         {
             var offset = 1;
             while (char.IsDigit(this.Peek(offset))) ++offset;
+            if (this.Peek(offset) == '.' && char.IsDigit(this.Peek(offset + 1)))
+            {
+                offset++;
+                while (char.IsDigit(this.Peek(offset))) ++offset;
+                var floatview = this.Advance(offset);
+                // TODO: Parsing into an float32 might not be the best idea
+                // TODO: Parsing for just dot is very cursed
+                var floatvalue = float.Parse(floatview.Span, NumberStyles.Any, new CultureInfo("en-US"));
+                this.tokenBuilder
+                    .SetType(TokenType.LiteralFloat)
+                    .SetText(floatview.ToString())
+                    .SetValue(floatvalue);
+                return default;
+            }
             var view = this.Advance(offset);
```

Comment:
A comment before this line would be nice that it's part of the regular integer lexing again